pymc/math.py

```diff
@@ -576,5 +527,13 @@ def block_diagonal(matrices, sparse=False, format="csr"):
     matrix
     """
     if len(matrices) == 1:  # graph optimization
+        warnings.warn(
```

Comment:
Any use of the function should issue a warning, not just this

Reply:
Alright!